app/apps/addresses/views.py

```diff
@@ -47,34 +46,41 @@ class AddressViewSet(ViewSet, GenericAPIView, PermitDetailsMixin):
         permission_classes=[permissions.CanAccessBRP],
     )
     def residents_by_bag_id(self, request, bag_id):
-        address = self.queryset.filter(bag_id=bag_id).first()
-        if not address:
+        # Get address
+        try:
+            address = Address.objects.get(bag_id=bag_id)
+        except Address.DoesNotExist:
+            address = Address(bag_id=bag_id)
+
+        # If no nummeraanduiding_id, get it!
+        if not address.nummeraanduiding_id:
+            try:
+                address.get_bag_address_data()
+                address.get_bag_nummeraanduiding_id()
+            except Exception:
+                return Response(
+                    {"error": "Bag data could not be obtained"},
+                    status=status.HTTP_404_NOT_FOUND,
+                )
```

Comment:
Generieke handler die voor iedere exceptie een 404 afgeeft is niet optimaal.
In principe wil je 404 alleen als iets niet gevonden wordt, en 500 als de server het verzoek niet kan afhandelen vanwege problemen.